libiconv 1.14 should conflict with newer compilers

@tgamblin 

### Expected Result

I expected libiconv 1.14 to work with gcc 5.3.0 and 6.2.0.

### Actual Result

The build failed and reported
```
../lib/.libs/libiconv.so: undefined reference to 'aliases2_lookup'
```
It worked with gcc 4.9.2.  According to the internet, the inline semantics in newer versions of gcc are causing this problem.  I have not tried libiconv 1.15 and do not know if this is an issue with that version as well.

### Steps to reproduce the issue

```console
spack install libiconv@1.14%gcc@5.3.0
spack install libiconv@1.14%gcc@6.2.0
```

--- a/var/spack/repos/builtin/packages/libiconv/package.py
+++ b/var/spack/repos/builtin/packages/libiconv/package.py
@@ -40,6 +40,8 @@ class Libiconv(AutotoolsPackage):
     # of C11 any more and thus might not exist.
     patch('gets.patch', when='@1.14')
 
+    conflicts('@1.14', when='%gcc@5:')
+
     def configure_args(self):
         args = ['--enable-extra-encodings']
 

